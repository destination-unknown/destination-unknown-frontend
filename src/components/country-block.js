import React from 'react'
import { StaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'
import styled from 'styled-components'

const LandContainer = styled.div`
  background: white;
  font-family: 'Open Sans', sans-serif;
`

const LandGridContainer = styled.div`
  display: grid;
  grid-template-columns: 42% 10% 3% 35%;
  margin: 0 auto;
  max-width: 1200px;
  padding: 64px 16px;
  @media only screen and (max-width: 600px) {
    grid-template-columns: 100%;
    grid-template-rows: 28% 1fr;
    padding: 16px;
  }
`

const LandSubTitle = styled.p`
  color: #121212;
  font-family: 'Lato', sans-serif;
  font-size: 3.5rem;
  font-weight: 900;
  margin: 0;
  @media only screen and (max-width: 600px) {
    font-size: 2.5rem;
  }
`

const LandImageContainer = styled.div`
  margin-top: 64px;
  grid-area: 1 / 1 / span 2 / span 3;
  @media only screen and (max-width: 600px) {
    margin-top: 16px;
    grid-area: 1 / 1;
  }
`

const LandTextContainer = styled.div`
  grid-area: 1 / 3 / auto / span 2;
  position: relative;
  @media only screen and (max-width: 600px) {
    grid-area: 2 / 1;
  }
`

const SubTitle = styled.p`
  color: #3b3b3b;
  margin-left: 80px;
  margin-top: 8px;
  line-height: 1.5;
  font-size: 1.1rem;
`

export const PureCountryBlock = ({ data }) => (
  <LandContainer>
    <LandGridContainer>
      <LandImageContainer>
        <Img fluid={data.landscapeImage.childImageSharp.fluid} />
      </LandImageContainer>
      <LandTextContainer>
        <LandSubTitle>195 LANDEN ZIJN ER IN DE WERELD</LandSubTitle>
        <SubTitle>
          Zoveel verschillende landen. Zoveel landen om uit te kiezen. Geen idee
          waar je naartoe wilt? Wees niet bang. Wij helpen je graag om een
          vakantieland te kiezen. Het enige wat jij hoeft te doen is de
          vragenlijst in te vullen. Wij doen de rest.
        </SubTitle>
      </LandTextContainer>
    </LandGridContainer>
  </LandContainer>
)

const CountryBlock = () => {
  return (
    <StaticQuery
      query={graphql`
        query {
          landscapeImage: file(relativePath: { eq: "landscape.jpg" }) {
            childImageSharp {
              fluid {
                ...GatsbyImageSharpFluid_noBase64
              }
            }
          }
        }
      `}
      render={data => <PureCountryBlock data={data} />}
    />
  )
}

export default CountryBlock
