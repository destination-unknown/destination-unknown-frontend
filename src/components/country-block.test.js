import React from 'react'
import { PureCountryBlock as CountryBlock } from './country-block'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const data = {
    "landImage": {
      "childImageSharp": {
        "fluid": {
          "base64": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAA7EAAAOxAGVKw4bAAAA+ElEQVQ4y+2STwqCQBSH5xpCRVFEOwk8R9iB2kQIhW08g5t2gqit6xadINtILSucfhOjqI0T/qFVDz58M858PN+TkH+0ERPXdTVKqQ5mDdA9z9MIkh1wwBEcGsDuO0x4AiFtJ0ISBMEcyRI8Mi9ijiyKZ57M824ikg5YgHtNIZOtQC+dDJcaICpciEtEyf4VmKkMSSpVFKWP9RacSy4X1xdg2bY9zP03WSliwKWRpKqkMgvnx0IPWyQbvFIj01MqGICZVJa9KxRKBiUcgFAoKlswqFtuAJ/t+i5VVZV9/hrswcb3/VElWcnhLpjyZ3WZrDfSfv1cWCdetnvGmXO4OtMAAAAASUVORK5CYII=",
          "tracedSVG": "data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' width='400' height='400' viewBox='0 0 400 400' version='1'%3e%3cpath d='M5 71H0v8l1 8 6-1c11 0 16-4 14-10s-3-6-16-5m377 2c-5 7 0 13 12 13l6 1V71h-6c-9-1-10-1-12 2M2 137c-2 0-2 1-2 9v10h10l1-7c1-7 0-11-4-12H2m391 0c-3 2-4 5-3 13l1 6h9v-10c0-7 0-9-2-9h-5M22 165c-2 0-2 0 0 3 4 7 13 14 14 11v-14H22m344 1v11l1 4 4-3 8-8 2-3-4-1h-11M47 190c-2 1 3 8 9 13 5 3 5 3 5 1v-14H47m294 1l-1 8 1 7 5-3 7-8 3-4h-15M72 215c-2 1 3 8 9 13 5 3 5 3 5 1v-14H72m244 1l-1 8 1 7 5-3 7-8 3-4h-15M97 240c-2 1 3 8 9 13 5 3 5 3 5 1v-14H97m194 1l-1 8 1 7 5-3 7-8 3-4h-15m-169 24c-2 1 3 8 9 13 5 3 5 3 5 1v-14h-14m144 1l-1 8 1 7 5-3 7-8 3-4h-15m-119 24c-2 0-1 3 3 7l1 2 5 4c4 3 5 3 5 1v-14h-14m94 1c-2 2 0 14 2 14l10-9 3-4-4-1h-11m-59 25l-9 3 6 8c8 6 33 7 42 0l6-6 2-4h-4l-10-1c-7-3-23-2-33 0' fill='lightgray' fill-rule='evenodd'/%3e%3c/svg%3e",
          "aspectRatio": 1,
          "src": "/static/51ef4d08e99f1e6f2046dd80a093815b/dd41c/caret-down.png",
          "srcSet": "/static/51ef4d08e99f1e6f2046dd80a093815b/dd41c/caret-down.png 16w",
          "srcWebp": "/static/51ef4d08e99f1e6f2046dd80a093815b/8b965/caret-down.webp",
          "srcSetWebp": "/static/51ef4d08e99f1e6f2046dd80a093815b/8b965/caret-down.webp 16w",
          "sizes": "(max-width: 16px) 100vw, 16px",
          "originalImg": "/static/51ef4d08e99f1e6f2046dd80a093815b/dd41c/caret-down.png",
          "originalName": "caret-down.png",
          "presentationWidth": 16,
          "presentationHeight": 16
        }
      }
    }
  }
  const tree = renderer.create(<CountryBlock data={data} />).toJSON()
  expect(tree).toMatchSnapshot()
})
