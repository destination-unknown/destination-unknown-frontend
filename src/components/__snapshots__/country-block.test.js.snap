// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa RgNsw"
>
  <div
    className="sc-bwzfXH hZYyYG"
  >
    <div
      className="sc-ifAKCX fmDQyT"
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "100%",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAA7EAAAOxAGVKw4bAAAA+ElEQVQ4y+2STwqCQBSH5xpCRVFEOwk8R9iB2kQIhW08g5t2gqit6xadINtILSucfhOjqI0T/qFVDz58M858PN+TkH+0ERPXdTVKqQ5mDdA9z9MIkh1wwBEcGsDuO0x4AiFtJ0ISBMEcyRI8Mi9ijiyKZ57M824ikg5YgHtNIZOtQC+dDJcaICpciEtEyf4VmKkMSSpVFKWP9RacSy4X1xdg2bY9zP03WSliwKWRpKqkMgvnx0IPWyQbvFIj01MqGICZVJa9KxRKBiUcgFAoKlswqFtuAJ/t+i5VVZV9/hrswcb3/VElWcnhLpjyZ3WZrDfSfv1cWCdetnvGmXO4OtMAAAAASUVORK5CYII="
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 0.5s",
              "transitionDelay": "0.25s",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' width='400' height='400' viewBox='0 0 400 400' version='1'%3e%3cpath d='M5 71H0v8l1 8 6-1c11 0 16-4 14-10s-3-6-16-5m377 2c-5 7 0 13 12 13l6 1V71h-6c-9-1-10-1-12 2M2 137c-2 0-2 1-2 9v10h10l1-7c1-7 0-11-4-12H2m391 0c-3 2-4 5-3 13l1 6h9v-10c0-7 0-9-2-9h-5M22 165c-2 0-2 0 0 3 4 7 13 14 14 11v-14H22m344 1v11l1 4 4-3 8-8 2-3-4-1h-11M47 190c-2 1 3 8 9 13 5 3 5 3 5 1v-14H47m294 1l-1 8 1 7 5-3 7-8 3-4h-15M72 215c-2 1 3 8 9 13 5 3 5 3 5 1v-14H72m244 1l-1 8 1 7 5-3 7-8 3-4h-15M97 240c-2 1 3 8 9 13 5 3 5 3 5 1v-14H97m194 1l-1 8 1 7 5-3 7-8 3-4h-15m-169 24c-2 1 3 8 9 13 5 3 5 3 5 1v-14h-14m144 1l-1 8 1 7 5-3 7-8 3-4h-15m-119 24c-2 0-1 3 3 7l1 2 5 4c4 3 5 3 5 1v-14h-14m94 1c-2 2 0 14 2 14l10-9 3-4-4-1h-11m-59 25l-9 3 6 8c8 6 33 7 42 0l6-6 2-4h-4l-10-1c-7-3-23-2-33 0' fill='lightgray' fill-rule='evenodd'/%3e%3c/svg%3e"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 0.5s",
              "transitionDelay": "0.25s",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 16px) 100vw, 16px"
            srcSet="/static/51ef4d08e99f1e6f2046dd80a093815b/8b965/caret-down.webp 16w"
            type="image/webp"
          />
          <source
            sizes="(max-width: 16px) 100vw, 16px"
            srcSet="/static/51ef4d08e99f1e6f2046dd80a093815b/dd41c/caret-down.png 16w"
          />
          <img
            alt=""
            onLoad={[Function]}
            src="/static/51ef4d08e99f1e6f2046dd80a093815b/dd41c/caret-down.png"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source type='image/webp' srcSet=\\"/static/51ef4d08e99f1e6f2046dd80a093815b/8b965/caret-down.webp 16w\\" sizes=\\"(max-width: 16px) 100vw, 16px\\" /><source srcSet=\\"/static/51ef4d08e99f1e6f2046dd80a093815b/dd41c/caret-down.png 16w\\" sizes=\\"(max-width: 16px) 100vw, 16px\\" /><img src=\\"/static/51ef4d08e99f1e6f2046dd80a093815b/dd41c/caret-down.png\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
    <div
      className="sc-EHOje bfmPNS"
    >
      <p
        className="sc-htpNat dzCKaP"
      >
        195
      </p>
    </div>
    <div
      className="sc-bZQynM cUCOVz"
    >
      <p
        className="sc-bxivhb imXitT"
      >
        landen zijn er in de wereld
      </p>
      <p
        className="sc-gzVnrw eWevwi"
      >
        Zoveel verschillende landen. Zoveel landen om uit te kiezen. Geen idee waar je naartoe wilt? Wees niet bang. Wij helpen je graag om een vakantieland te kiezen. Het enige wat jij hoeft te doen is de vragenlijst in te vullen. Wij doen de rest.
      </p>
    </div>
  </div>
</div>
`;
